core/templates/services/questions-list.service.ts

```diff
@@ -57,13 +58,21 @@ export class QuestionsListService {
     this._questionSummartiesInitializedEventEmitter.emit();
   }
 
-  private _setNextQuestionsCursor(nextCursor: string): void {
-    this._nextCursorForQuestions = nextCursor;
+  private _changeNextQuestionsOffset(
+      nextOffset: number, resetHistory: boolean): void {
+    if (resetHistory) {
+      this._nextOffsetForQuestions = 0;
+    }
+    this._nextOffsetForQuestions += nextOffset;
```

Comment:
Why are you adding nextOffset rather than the number of questions fetched in this batch? That doesn't make sense conceptually.

Reply:
Yeah, shouldn't a constant be added here which is basically the number of questions in the page?

Reply:
`cosntants.NUM_QUESTIONS_PER_PAGE` can be used

Reply:
Now it is using `constants.NUM_QUESTIONS_PER_PAGE`. Done.